raft-tech/TANF-app
# 2954 - Remove SESSION_TIMEOUT

## Summary of Changes
Pull request closes #2954 
- removes the old/irrelevant `SESSION_TIMEOUT` which hasn't controlled the session timeout since #2911 
- ~~extend~~ `SESSION_COOKIE_AGE` to match the expiration of the jwt provided by login.gov - the expiration in the dev environment was only 15 minutes
- Update the documentation to reflect the change in session management

## How to Test
```
cd tdrs-backend && docker-compose up
cd tdrs-frontend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in.

That's basically it, nothing changes about our signin workflow. The values set by `SESSION_TIMEOUT` were being overriden by `SESSION_EXPIRE_AT_BROWSER_CLOSE`'s behavior of removing the `Expires=` from the cookie. You can run the tests from #2911 that verify the session cookie behavior.

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Identify session timeout/expiration for the jwt provided by login.gov
+ [x] Parameterize environment variable value for `SESSION_COOKIE_AGE` (match or lower than login.gov jwt expiration)
+ [x] Remove unused `SESSION_TIMEOUT` variable in `common.py` and usages throughout the authentication api

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
